### DISCUSSION TITLE
how to mock delay expressions ?

### DISCUSSION THREAD

**Discussion by ilovett** [statelyai/xstate#1840] [Q&A] [answered]
My state machine uses as delay expression: https://xstate.js.org/docs/guides/delays.html#delay-expressions-on-transitions

What is the best way to mock this time as elapsing or expression completing?

For most other async related code I have relied on flushing promises like so:

```
function flushPromises() {
  return new Promise((resolve) => setImmediate(resolve));
}
```

and in tests

```
// send an action

await flushPromises();

// verify state
```

I'm not really sure how to mock this though, is there some exact action I need to send or something else?  What's the best way to do this?

**Comment by ilovett** [comment, marked as answer]
My bad.  I just found the `SimulatedClock`.

**Reply by davidkpiano** [reply]
Relevant part of the docs: https://xstate.js.org/docs/guides/delays.html#interpretation
